v2/inc/c_logging/log_context.h

```diff
@@ -0,0 +1,105 @@
+#define LOCAL_LOG_CONTEXT_DEFINE(log_context, parent_context, ...) \
+    char MU_C2(log_string_, log_context)[LOG_MAX_MESSAGE_LENGTH]; \
+    uint32_t MU_C2(parent_context_string_length_, log_context) = (uint32_t)log_context_get_string_length(parent_context); \
+    { \
+        const char* parent_context_string = log_context_get_string(parent_context); \
+        if (parent_context_string != NULL) \
+        { \
+            (void)memcpy(MU_C2(log_string_, log_context), parent_context_string, MU_C2(parent_context_string_length_, log_context)); \
+        } \
+        char* pos = MU_C2(log_string_, log_context) + MU_C2(parent_context_string_length_, log_context); \
+        int chars_written = 0; \
+        MU_FOR_EACH_1(STRINGIFY_LOG_CONTEXT_FIELD, __VA_ARGS__) \
+    } \
+    LOG_CONTEXT log_context = { MU_C2(log_string_, log_context) };
+
+#define EXPAND_GET_LENGTH_LOG_CONTEXT_FIELD(field_name, field_PRI, field_values) \
+    chars_written += snprintf(NULL, 0, "%s=" field_PRI " ", field_name, field_values); \
+
+#define GET_LOG_CONTEXT_FIELD_LENGTH(field_desc) \
+    MU_C2(EXPAND_GET_LENGTH_, field_desc)
+
+static size_t log_context_get_string_length(LOG_CONTEXT_HANDLE log_context)
+{
+    return (log_context == NULL) ? 0 : strlen(log_context->context_string);
+}
+
+static const char* log_context_get_string(LOG_CONTEXT_HANDLE log_context)
+{
+    return (log_context == NULL) ? NULL : log_context->context_string;
+}
+
+// macro that can be used to create a dynamically allocated context
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    uint32_t MU_C2(parent_context_string_length_, dest_log_context) = (uint32_t)log_context_get_string_length(parent_context); \
+    int MU_C2(chars_to_write_, dest_log_context); \
+    { \
+        int chars_written = 0; \
+        MU_FOR_EACH_1(GET_LOG_CONTEXT_FIELD_LENGTH, __VA_ARGS__) \
+        MU_C2(chars_to_write_, dest_log_context) = chars_written + MU_C2(parent_context_string_length_, dest_log_context) + 1; \
+    } \
+    dest_log_context = malloc(sizeof(LOG_CONTEXT) + MU_C2(chars_to_write_, dest_log_context) + 1); \
+    if (dest_log_context == NULL) \
+    { \
+        (void)printf("malloc(sizeof(LOG_CONTEXT) + %d + 1) failed", MU_C2(chars_to_write_, dest_log_context)); \
+    } \
+    else \
+    { \
+        dest_log_context->context_string = (void*)((LOG_CONTEXT*)(dest_log_context) + 1); \
+        { \
+            char* pos = dest_log_context->context_string + MU_C2(parent_context_string_length_, dest_log_context); \
+            const char* parent_context_string = log_context_get_string(parent_context); \
+            if (parent_context_string != NULL) \
+            { \
+                (void)memcpy(dest_log_context->context_string, parent_context_string, MU_C2(parent_context_string_length_, dest_log_context)); \
+            } \
+            int chars_written = 0; \
+            MU_FOR_EACH_1(STRINGIFY_LOG_CONTEXT_FIELD, __VA_ARGS__) \
```

Comment:
>1(STRINGIFY_LOG_CONTEXT_FIELD, __VA_ARGS__) \ [](http://example.com/codeflow?start=24&length=45)

this seems to produce a concatenation of "parent" and "child" context strings, how does one where the parent ended and the child began in the concatenated string? [also seeing that the context is an unstructured data]

Reply:
Right now there is no separation.
With the next version of what I did I believe we could add that.

Reply:
This will need more work. But I think I might try it.

Reply:
Structure has been added to console and etw